tests/test_clean_collections.py

```diff
@@ -0,0 +1,204 @@
+from lsst.daf.butler.tests import MetricsExample, addDataIdValue, addDatasetType, registerMetricsExample
+
+
+class CleanCollectionsTestCase(unittest.IsolatedAsyncioTestCase):
+
+    def setUp(self):
+        """set up test info, including config dict, and populate
+        Butler repo with test data
+        """
+
+        config_name = "clean_collections.yaml"
+
+        # create a path to the configuration file
+
+        testdir = os.path.abspath(os.path.dirname(__file__))
+        configFile = os.path.join(testdir, "etc", config_name)
+
+        # load the YAML configuration
+
+        with open(configFile, "r") as f:
+            self.config = yaml.safe_load(f)
+
+        # extract parts of the ingester configuration
+        # and alter the image staging directory to point
+        # at the temporary directories created for his test
+
+        ingesterConfig = self.config["ingester"]
+        self.imageDir = tempfile.mkdtemp()
+        ingesterConfig["imageStagingDirectory"] = self.imageDir
+
+        self.badDir = tempfile.mkdtemp()
+        butlerConfig = ingesterConfig["butlers"][0]["butler"]
+        butlerConfig["badFileDirectory"] = self.badDir
+        self.stagingDir = tempfile.mkdtemp()
+        butlerConfig["stagingDirectory"] = self.stagingDir
+
+        self.repoDir = tempfile.mkdtemp()
+        Butler.makeRepo(self.repoDir)
+
+        butlerConfig["repoDirectory"] = self.repoDir
+
+        self.clean_collections = butlerConfig["cleanCollections"]
+
+        # Define the run collection
+        run_a = "collection_a"
+        run_b = "collection_b"
+        self.collections = [run_a, run_b]
+
+        # Initialize the Butler
+        opts = dict(writeable=True)
+        self.butler = Butler(self.repoDir, **opts)
+
+        self.butler.registry.registerCollection("collection_a", CollectionType.RUN)
+        self.butler.registry.registerCollection("collection_b", CollectionType.RUN)
+
+        registerMetricsExample(self.butler)
+        image_data = MetricsExample(summary={"answer": 42, "question": "unknown"})
+        image_data2 = MetricsExample(summary={"answer": 42, "question": "unknown"})
+
+        # Define dataset type and data ID
+        data_id_1 = {"instrument": "notACam", "visit": 12345, "detector": 1}
+        data_id_2 = {"instrument": "notACam", "visit": 12346, "detector": 1}
+
+        addDataIdValue(self.butler, "instrument", "notACam")
+        addDatasetType(self.butler, "DataType1", {"instrument"}, "StructuredDataNoComponents")
+        addDatasetType(self.butler, "DataType2", {"instrument"}, "StructuredDataNoComponents")
+
+        # Write the image to the Butler under the run collection
+        self.butler.put(image_data, "DataType1", dataId=data_id_1, run=run_a)
+        self.butler.put(image_data2, "DataType2", dataId=data_id_2, run=run_b)
+
+    def tearDown(self):
+        """Remove butler repo directory"""
+        shutil.rmtree(self.repoDir, ignore_errors=True)
+
+    def number_of_files(self):
```

Comment:
It's a bit nitpicky, but this functions really counts number of _datasets_ in the Butler repo. Generally speaking, number of datasets may not correspond to actual number of files so `number_of_datasets` could be potentially more apt name for it.

Reply:
Agreed.  Changed.